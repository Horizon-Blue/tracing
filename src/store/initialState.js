export default {
  locale: 'en',
  token: null,
};
