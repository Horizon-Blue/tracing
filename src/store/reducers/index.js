import { combineReducers } from 'redux';
import locale from './locale';
import token from './token';

export default combineReducers({ locale, token });
