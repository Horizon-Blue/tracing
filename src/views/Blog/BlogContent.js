import React, { PureComponent } from 'react';
import markdownIt from 'markdown-it';
import mk from 'markdown-it-katex';
import prism from 'markdown-it-prism';
import { Container } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import './prism-atom-dark.css';

const md = markdownIt({ html: true, typographer: true, xhtmlOut: true })
  .use(mk)
  .use(prism);

/**
 * The component that renders blog post content (Markdown string) into
 * DOM nodes
 *
 * @class      BlogContent (name)
 */
class BlogContent extends PureComponent {
  static propTypes = {
    content: PropTypes.string.isRequired,
  };

  /**
   * Helper function to render content
   *
   * @param      {<type>}  text    The text
   * @return     {<type>}  { description_of_the_return_value }
   */
  static renderPost(text) {
    return md.render(text);
  }

  render() {
    return (
      <Container text textAlign="left">
        <main
          dangerouslySetInnerHTML={{
            __html: BlogContent.renderPost(this.props.content),
          }}
        />
      </Container>
    );
  }
}

export default BlogContent;
