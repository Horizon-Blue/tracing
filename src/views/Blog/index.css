section.blog-list {
  padding: 5vh 5vw;
  border-bottom: 1px solid rgb(35, 36, 45);
  transition: background-color 500ms ease-in, color 500ms ease-in;
}

section.blog-list:hover {
  background-color: rgba(12, 96, 170, 0.3);
}

.ui.basic.blue.label {
  background-color: rgba(14, 42, 118, 0.4) !important;
  font-size: 0.9em;
}

section.blog-list .tag-container {
  margin-top: 1rem !important;
}

@media only screen and (min-width: 992px) {
  section.blog-list .tag-container {
    text-align: right;
    margin-top: 0 !important;
  }
}

section.blog-list summary {
  color: rgba(157, 204, 254, 0.8);
  font-size: 1.1em;
}

section.blog-list .blog-meta-divider {
  margin: 0 0.5rem;
}

article.blog-post {
  padding-top: 3vh;
  color: white;
}

article.blog-post .blog-post-meta {
  color: rgba(157, 204, 254, 0.8);
}

article.blog-post .blog-post-meta span:not(:first-child) {
  margin-left: 1vw;
}

article.blog-post main {
  margin: 2vh 0 5vh 0;
}
