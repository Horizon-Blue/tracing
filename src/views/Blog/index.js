import React, { PureComponent } from 'react';
import { Route, Switch } from 'react-router-dom';
import BlogList from './BlogList';
import BlogPost from './BlogPost';
import './index.less';

/**
 * The Blog Section.
 *
 * @class      Blog (name)
 */
class Blog extends PureComponent {
  render() {
    return (
      <Switch>
        <Route exact path="/blog" component={BlogList} />
        <Route path="/blog/:postId" component={BlogPost} />
      </Switch>
    );
  }
}

export default Blog;
