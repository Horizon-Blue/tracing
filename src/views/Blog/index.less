section.blog-list {
  padding: 5vh 5vw;
  border-bottom: 1px solid rgb(35, 36, 45);
  transition: background-color 500ms ease-in, color 500ms ease-in;

  &:hover {
    background-color: rgba(12, 96, 170, 0.3);
  }

  .tag-container {
    margin-top: 1rem !important;
    @media (min-width: 992px) {
      text-align: right;
      margin-top: 0 !important;
    }
  }

  summary {
    color: rgba(157, 204, 254, 0.8);
    font-size: 1.1em;
  }

  .blog-meta-divider {
    margin: 0 0.5rem;
  }
}

.ui.basic.blue.label {
  background-color: rgba(14, 42, 118, 0.4) !important;
  font-size: 0.9em;
}

form.search-box {
  margin-top: 2vh;
  margin-left: 5vw;
  border-bottom: 1px solid rgb(35, 36, 45);
}

article.blog-post {
  padding-top: 3vh;
  color: white;

  h2,
  p,
  .comment-username {
    color: white !important;
  }

  .blog-post-meta,
  .comment-meta,
  .comment-action a {
    color: rgba(157, 204, 254, 0.8) !important;
  }

  .blog-post-meta > span:not(:first-child) {
    margin-left: 1vw;
  }

  main {
    margin: 2vh 0 5vh 0;
  }

  .comment-container {
    margin-bottom: 1vh !important;
    padding: 1vh 1vh !important;
    transition: background-color 500ms ease-in, color 500ms ease-in;

    &:hover {
      background-color: rgba(12, 96, 170, 0.3);
    }
  }
}
