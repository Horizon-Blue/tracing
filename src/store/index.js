import { createStore } from 'redux';
import initialState from './initialState';
import rootReducer from './reducers';

const store = createStore(
  rootReducer,
  initialState,
  /* Enable Redux Devtools Chrome Extension */
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
