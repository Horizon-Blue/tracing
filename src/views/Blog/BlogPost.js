import React, { PureComponent } from 'react';
import {
  Container,
  Header,
  Icon,
  Divider,
  Label,
  Grid,
  Loader,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import BlogContent from './BlogContent';
import Translated from '../Translated';
import Comments from './Comments';

/**
 * The content to show on blog post page (for each post)
 *
 * @class      BlogPost (name)
 */
class BlogPost extends PureComponent {
  static propTypes = {
    data: PropTypes.object.isRequired,
  };

  /**
   * Render each tag as a DOM node
   *
   * @param      {String}  name    The name of the tag
   * @return     {Node}  The Node corresponds to the tag
   */
  renderTag({ node: tag }) {
    return (
      <Label basic color="blue" horizontal key={tag.id}>
        {tag.name}
      </Label>
    );
  }

  render() {
    const { data: { loading, error, post } } = this.props;

    return loading ? (
      <Loader inline="centered" size="big" inverted active={loading} />
    ) : !post ? (
      <div>{error}</div>
    ) : (
      <Container as="article" className="blog-post">
        <Container text as="section" textAlign="center">
          <Header as="h1" inverted>
            {post.title}
          </Header>
          <div className="blog-post-meta">
            <span>
              <Icon name="user outline" />
              {post.author.name}
            </span>
            <span>
              <Icon name="calendar outline" />
              {post.publishDate.slice(0, 10)}
            </span>
          </div>
        </Container>
        <BlogContent content={post.content} />
        <Divider inverted />
        <Container as="section">
          <Grid inverted divided padded columns="equal">
            <Grid.Row>
              <Grid.Column>
                <Translated id="tags" />: {post.tags.edges.map(this.renderTag)}
              </Grid.Column>
              <Grid.Column>
                <Translated id="category" />: {post.category.name}
              </Grid.Column>
              <Grid.Column>
                <Translated id="updatedAt" />:{' '}
                {post.lastUpdateDate.slice(0, 10)}
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Container>
        <Divider inverted />
        <Comments comments={post.comments.edges} />
      </Container>
    );
  }
}

const getPost = gql`
  query($postId: ID!) {
    post(id: $postId) {
      id
      title
      category {
        id
        name
      }
      tags {
        edges {
          node {
            id
            name
          }
        }
      }
      author {
        id
        name
      }
      publishDate
      lastUpdateDate
      content
      comments {
        edges {
          node {
            id
            author {
              id
              name
              avatar
            }
            createDate
            content
          }
        }
      }
    }
  }
`;

export { BlogPost as BlogPostView };

export default graphql(getPost, {
  options: ({ match }) => ({ variables: match.params }),
})(BlogPost);
