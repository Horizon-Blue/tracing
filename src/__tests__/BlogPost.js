import React from 'react';
import { BlogPostView } from '../views/Blog/BlogPost';
import renderer from 'react-test-renderer';
import store from '../store';
import { Provider } from 'react-redux';

it('renders loading', () => {
  const tree = renderer
    .create(<BlogPostView data={{ loading: true }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const post = {
  id: 1234,
  title: 'Foo Bar',
  author: {
    name: 'Xiaoyan',
  },
  href: 'foo-bar',
  publishDate: '2017-06-26T23:20:11',
  lastUpdateDate: '2017-08-23T10:20:11',
  tags: {
    edges: [{ node: { id: 123, name: 'Test' } }],
  },
  category: {
    id: 'foo',
    name: 'journal',
  },
  comments: {
    edges: [],
  },
  content: `
## This is a Markdown Test

See if this can be *rendered* **correctly**

- This one
- And This one

![test img](/favicon.png)

### Inline HTML

<i style="color:orange;">I should be orange</i>

<iframe width="300" height="200"
src="https://www.youtube-nocookie.com/embed/ZXsQAXx_ao0?rel=0"
frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>

### LaTeX

$$\\frac{1}{N}\\sum_{n=1}^{N}e^{n}$$

### Code Block

\`\`\`python
import antigravity

print("Hello, world")
\`\`\`

\`\`\`jsx
import React, { PureComponent } from 'react';
\`\`\`

## Jupyter Notebook

I didn't find any good Jupyter Notebook rendering library.
However, exporting Jupyter Notebook as Markdown works fairly
weel and renders properly.

e.g.

Gradients
---------
let's backprop now
\`\`out.backward()\`\` is equivalent to doing \`\`out.backward(torch.Tensor([1.0]))\`\`




\`\`\`python
out.backward()
\`\`\`

print gradients $\\frac{\\partial out}{\\partial x}$
`,
};

it('render with content', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BlogPostView data={{ loading: false, post }} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
