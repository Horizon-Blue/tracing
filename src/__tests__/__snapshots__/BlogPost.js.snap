// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with content 1`] = `
<article
  className="ui container blog-post"
>
  <section
    className="ui text center aligned container"
  >
    <h1
      className="ui inverted header"
    >
      Foo Bar
    </h1>
    <div
      className="blog-post-meta"
    >
      <span>
        <i
          aria-hidden="true"
          className="user outline icon"
        />
        Xiaoyan
      </span>
      <span>
        <i
          aria-hidden="true"
          className="calendar outline icon"
        />
        2017-06-26
      </span>
    </div>
  </section>
  <div
    className="ui text left aligned container"
  >
    <main
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h2>This is a Markdown Test</h2>
        <p>See if this can be <em>rendered</em> <strong>correctly</strong></p>
        <ul>
        <li>This one</li>
        <li>And This one</li>
        </ul>
        <p><img src=\\"/favicon.png\\" alt=\\"test img\\" /></p>
        <h3>Inline HTML</h3>
        <p><i style=\\"color:orange;\\">I should be orange</i></p>
        <iframe width=\\"300\\" height=\\"200\\"
        src=\\"https://www.youtube-nocookie.com/embed/ZXsQAXx_ao0?rel=0\\"
        frameborder=\\"0\\" allow=\\"autoplay; encrypted-media\\" allowfullscreen></iframe>
        <h3>LaTeX</h3>
        <p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math><semantics><mrow><mfrac><mrow><mn>1</mn></mrow><mrow><mi>N</mi></mrow></mfrac><msubsup><mo>∑</mo><mrow><mi>n</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></msubsup><msup><mi>e</mi><mrow><mi>n</mi></mrow></msup></mrow><annotation encoding=\\"application/x-tex\\">\\\\frac{1}{N}\\\\sum_{n=1}^{N}e^{n}
        </annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"strut\\" style=\\"height:1.8283360000000002em;\\"></span><span class=\\"strut bottom\\" style=\\"height:3.0954490000000003em;vertical-align:-1.267113em;\\"></span><span class=\\"base displaystyle textstyle uncramped\\"><span class=\\"mord reset-textstyle displaystyle textstyle uncramped\\"><span class=\\"sizing reset-size5 size5 reset-textstyle textstyle uncramped nulldelimiter\\"></span><span class=\\"mfrac\\"><span class=\\"vlist\\"><span style=\\"top:0.686em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span class=\\"reset-textstyle textstyle cramped\\"><span class=\\"mord textstyle cramped\\"><span class=\\"mord mathit\\" style=\\"margin-right:0.10903em;\\">N</span></span></span></span><span style=\\"top:-0.22999999999999998em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span class=\\"reset-textstyle textstyle uncramped frac-line\\"></span></span><span style=\\"top:-0.677em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span class=\\"reset-textstyle textstyle uncramped\\"><span class=\\"mord textstyle uncramped\\"><span class=\\"mord mathrm\\">1</span></span></span></span><span class=\\"baseline-fix\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span>​</span></span></span><span class=\\"sizing reset-size5 size5 reset-textstyle textstyle uncramped nulldelimiter\\"></span></span><span class=\\"mop op-limits\\"><span class=\\"vlist\\"><span style=\\"top:1.1671129999999998em;margin-left:0em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span class=\\"reset-textstyle scriptstyle cramped\\"><span class=\\"mord scriptstyle cramped\\"><span class=\\"mord mathit\\">n</span><span class=\\"mrel\\">=</span><span class=\\"mord mathrm\\">1</span></span></span></span><span style=\\"top:-0.000005000000000143778em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span><span class=\\"op-symbol large-op mop\\">∑</span></span></span><span style=\\"top:-1.2500050000000003em;margin-left:0em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span class=\\"reset-textstyle scriptstyle uncramped\\"><span class=\\"mord scriptstyle uncramped\\"><span class=\\"mord mathit\\" style=\\"margin-right:0.10903em;\\">N</span></span></span></span><span class=\\"baseline-fix\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span>​</span></span></span><span class=\\"mord\\"><span class=\\"mord mathit\\">e</span><span class=\\"vlist\\"><span style=\\"top:-0.413em;margin-right:0.05em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span class=\\"reset-textstyle scriptstyle uncramped\\"><span class=\\"mord scriptstyle uncramped\\"><span class=\\"mord mathit\\">n</span></span></span></span><span class=\\"baseline-fix\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span>​</span></span></span></span></span></span></span></p>
        <h3>Code Block</h3>
        <pre class=\\"language-python\\"><code class=\\"language-python\\"><span class=\\"token keyword\\">import</span> antigravity
        
        <span class=\\"token keyword\\">print</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">\\"Hello, world\\"</span><span class=\\"token punctuation\\">)</span>
        </code></pre>
        <pre class=\\"language-jsx\\"><code class=\\"language-jsx\\"><span class=\\"token keyword\\">import</span> React<span class=\\"token punctuation\\">,</span> <span class=\\"token punctuation\\">{</span> PureComponent <span class=\\"token punctuation\\">}</span> <span class=\\"token keyword\\">from</span> <span class=\\"token string\\">'react'</span><span class=\\"token punctuation\\">;</span>
        </code></pre>
        <h2>Jupyter Notebook</h2>
        <p>I didn’t find any good Jupyter Notebook rendering library.
        However, exporting Jupyter Notebook as Markdown works fairly
        weel and renders properly.</p>
        <p>e.g.</p>
        <h2>Gradients</h2>
        <p>let’s backprop now
        <code>out.backward()</code> is equivalent to doing <code>out.backward(torch.Tensor([1.0]))</code></p>
        <pre class=\\"language-python\\"><code class=\\"language-python\\">out<span class=\\"token punctuation\\">.</span>backward<span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span>
        </code></pre>
        <p>print gradients <span class=\\"katex\\"><span class=\\"katex-mathml\\"><math><semantics><mrow><mfrac><mrow><mi mathvariant=\\"normal\\">∂</mi><mi>o</mi><mi>u</mi><mi>t</mi></mrow><mrow><mi mathvariant=\\"normal\\">∂</mi><mi>x</mi></mrow></mfrac></mrow><annotation encoding=\\"application/x-tex\\">\\\\frac{\\\\partial out}{\\\\partial x}</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"strut\\" style=\\"height:0.8801079999999999em;\\"></span><span class=\\"strut bottom\\" style=\\"height:1.2251079999999999em;vertical-align:-0.345em;\\"></span><span class=\\"base textstyle uncramped\\"><span class=\\"mord reset-textstyle textstyle uncramped\\"><span class=\\"sizing reset-size5 size5 reset-textstyle textstyle uncramped nulldelimiter\\"></span><span class=\\"mfrac\\"><span class=\\"vlist\\"><span style=\\"top:0.345em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span class=\\"reset-textstyle scriptstyle cramped\\"><span class=\\"mord scriptstyle cramped\\"><span class=\\"mord mathrm\\" style=\\"margin-right:0.05556em;\\">∂</span><span class=\\"mord mathit\\">x</span></span></span></span><span style=\\"top:-0.22999999999999998em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span class=\\"reset-textstyle textstyle uncramped frac-line\\"></span></span><span style=\\"top:-0.394em;\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span><span class=\\"reset-textstyle scriptstyle uncramped\\"><span class=\\"mord scriptstyle uncramped\\"><span class=\\"mord mathrm\\" style=\\"margin-right:0.05556em;\\">∂</span><span class=\\"mord mathit\\">o</span><span class=\\"mord mathit\\">u</span><span class=\\"mord mathit\\">t</span></span></span></span><span class=\\"baseline-fix\\"><span class=\\"fontsize-ensurer reset-size5 size5\\"><span style=\\"font-size:0em;\\">​</span></span>​</span></span></span><span class=\\"sizing reset-size5 size5 reset-textstyle textstyle uncramped nulldelimiter\\"></span></span></span></span></span></p>
        ",
        }
      }
    />
  </div>
  <div
    className="ui inverted divider"
  />
  <section
    className="ui container"
  >
    <div
      className="ui inverted divided padded equal width grid"
    >
      <div
        className="row"
      >
        <div
          className="column"
        >
          <span
            dispatch={undefined}
          >
            Tags
          </span>
          : 
          <div
            className="ui blue basic horizontal label"
            onClick={[Function]}
          >
            Test
          </div>
        </div>
        <div
          className="column"
        >
          <span
            dispatch={undefined}
          >
            Category
          </span>
          : 
          journal
        </div>
        <div
          className="column"
        >
          <span
            dispatch={undefined}
          >
            Last update
          </span>
          :
           
          2017-08-23
        </div>
      </div>
    </div>
  </section>
  <div
    className="ui inverted divider"
  />
  <div
    className="ui text container blog-comments"
  >
    <div
      className="ui large comments"
    >
      <h2
        className="ui dividing header"
        dispatch={undefined}
      >
        Comments
      </h2>
    </div>
  </div>
</article>
`;

exports[`renders loading 1`] = `
<div
  className="ui big active inverted centered inline loader"
/>
`;
